smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/lang/KotlinTypes.kt

```diff
@@ -38,6 +38,8 @@ object KotlinTypes {
         val List: Symbol = builtInSymbol("List", "kotlin.collections")
         val Set: Symbol = builtInSymbol("Set", "kotlin.collections")
         val Map: Symbol = builtInSymbol("Map", "kotlin.collections")
+        val mutableListOf: Symbol = builtInSymbol("mutableListOf", "kotlin.collections")
```

Comment:
Are these really necessary?

Reply:
Technically no. This was mostly left over from an abandoned attempt to fully genericize symbols inside the `KotlinSymbolProvider`. Could remove and simplify the deserialization codegen if you like.

Reply:
No strong feelings, it's fine as is.